.. documentation master file, created by sphinx-quickstart
   You can adapt this file completely to your liking, but it should at least
   contain the root `toctree` directive.

.. raw:: html

    <style> .red {color:red} </style>

.. role:: red

BIT - Bayesian Identification of Transcriptional Regulators from Epigenomics-Based Query Region Sets
===========================================================


.. image:: ./images/Figure_1.png

**BIT** (Bayesian Identification of Transcriptional Regulators from Epigenomic-Based Query Region Sets) is a novel bioinformatics tool designed to identify transcriptional regulators from a user-provided set of epigenomic regions. These regions are typically derived from genome-wide epigenomic profiling data that probe specific biological processes, such as ATAC-seq and DNase-seq. Using a pre-processed reference library containing over 10,000 human TR ChIP-seq datasets and over 5,000 mouse TR ChIP-seq datasets, **BIT** leverages a Bayesian hierarchical model to achieve higher accuracy and enhanced interpretability.


`GitHub <https://github.com/ZeyuL01/BIT>`_

Online portal can be accessed from: `BIT Portal <http://43.135.174.109:8080/>`_

If you use BIT in your work, please cite us:

.. toctree::
   :maxdepth: 1
   :caption: BIT
   :name: BIT

   InstallationGuide
   QuickStart
   Examples/index
   Functions
   OnlinePortal

.. toctree::
   :maxdepth: 1
   :caption: Appendix
   :name: Appendix

   BIT's paper <https://www.biorxiv.org/content/10.1101/2024.06.02.597061v2>
   References
   Dr. Xinlei (Sherry) Wang <https://www.uta.edu/academics/faculty/profile?username=wangx9>
   Dr. Lin Xu <https://qbrc.swmed.edu/labs/xulab/>
