References
==========

Here are some important references used in this project:

- 1	Casamassimi, A. & Ciccodicola, A. Transcriptional Regulation: Molecules, Involved Mechanisms, and Misregulation. Int J Mol Sci 20 (2019). https://doi.org/10.3390/ijms20061281

- 2	Andersson, R. & Sandelin, A. Determinants of enhancer and promoter activities of regulatory elements. Nat Rev Genet 21, 71-87 (2020). https://doi.org/10.1038/s41576-019-0173-8

- 3	Mullen, A. C. et al. Master transcription factors determine cell-type-specific responses to TGF-beta signaling. Cell 147, 565-576 (2011). https://doi.org/10.1016/j.cell.2011.08.050

- 4	Whyte, W. A. et al. Master transcription factors and mediator establish super-enhancers at key cell identity genes. Cell 153, 307-319 (2013). https://doi.org/10.1016/j.cell.2013.03.035

- 5	Hwang, J. Y. & Zukin, R. S. REST, a master transcriptional regulator in neurodegenerative disease. Curr Opin Neurobiol 48, 193-200 (2018). https://doi.org/10.1016/j.conb.2017.12.008

- 6	Bradner, J. E., Hnisz, D. & Young, R. A. Transcriptional Addiction in Cancer. Cell 168, 629-643 (2017). https://doi.org/10.1016/j.cell.2016.12.013

- 7	Chen, Y., Xu, L., Lin, R. Y., Muschen, M. & Koeffler, H. P. Core transcriptional regulatory circuitries in cancer. Oncogene 39, 6633-6646 (2020). https://doi.org/10.1038/s41388-020-01459-w

- 8	Pei, H., Guo, W., Peng, Y., Xiong, H. & Chen, Y. Targeting key proteins involved in transcriptional regulation for cancer therapy: Current strategies and future prospective. Med Res Rev 42, 1607-1660 (2022). https://doi.org/10.1002/med.21886

- 9	Rottner, A. K. et al. A genome-wide CRISPR screen identifies CALCOCO2 as a regulator of beta cell function influencing type 2 diabetes risk. Nat Genet 55, 54-65 (2023). https://doi.org/10.1038/s41588-022-01261-2

- 10	Balakrishnan, S., Dhavamani, S. & Prahalathan, C. beta-Cell specific transcription factors in the context of diabetes mellitus and beta-cell regeneration. Mech Dev 163, 103634 (2020). https://doi.org/10.1016/j.mod.2020.103634

- 11	Hong, J. H. & Zhang, H. G. Transcription Factors Involved in the Development and Prognosis of Cardiac Remodeling. Front Pharmacol 13, 828549 (2022). https://doi.org/10.3389/fphar.2022.828549

- 12	Bauer, A. J. & Martin, K. A. Coordinating Regulation of Gene Expression in Cardiovascular Disease: Interactions between Chromatin Modifiers and Transcription Factors. Front Cardiovasc Med 4, 19 (2017). https://doi.org/10.3389/fcvm.2017.00019

- 13	Bhagwat, A. S. & Vakoc, C. R. Targeting Transcription Factors in Cancer. Trends Cancer 1, 53-65 (2015). https://doi.org/10.1016/j.trecan.2015.07.001

- 14	Bushweller, J. H. Targeting transcription factors in cancer - from undruggable to reality. Nat Rev Cancer 19, 611-624 (2019). https://doi.org/10.1038/s41568-019-0196-7

- 15	Darnell, J. E., Jr. Transcription factors as targets for cancer therapy. Nat Rev Cancer 2, 740-749 (2002). https://doi.org/10.1038/nrc906

- 16	Qin, Q. et al. Lisa: inferring transcriptional regulators through integrative modeling of public chromatin accessibility and ChIP-seq data. Genome Biol 21, 32 (2020). https://doi.org/10.1186/s13059-020-1934-6

- 17	Wang, Z. et al. BART: a transcription factor prediction tool with query gene sets or epigenomic profiles. Bioinformatics 34, 2867-2869 (2018). https://doi.org/10.1093/bioinformatics/bty194

- 18	Keenan, A. B. et al. ChEA3: transcription factor enrichment analysis by orthogonal omics integration. Nucleic Acids Res 47, W212-W224 (2019). https://doi.org/10.1093/nar/gkz446

- 19	Puente-Santamaria, L., Wasserman, W. W. & Del Peso, L. TFEA.ChIP: a tool kit for transcription factor binding site enrichment analysis capitalizing on ChIP-seq datasets. Bioinformatics 35, 5339-5340 (2019). https://doi.org/10.1093/bioinformatics/btz573

- 20	Roopra, A. MAGIC: A tool for predicting transcription factors and cofactors driving gene sets using ENCODE data. PLoS Comput Biol 16, e1007800 (2020). https://doi.org/10.1371/journal.pcbi.1007800

- 21	Machanick, P. & Bailey, T. L. MEME-ChIP: motif analysis of large DNA datasets. Bioinformatics 27, 1696-1697 (2011). https://doi.org/10.1093/bioinformatics/btr189

- 22	Heinz, S. et al. Simple combinations of lineage-determining transcription factors prime cis-regulatory elements required for macrophage and B cell identities. Mol Cell 38, 576-589 (2010). https://doi.org/10.1016/j.molcel.2010.05.004

- 23	Tanigawa, Y., Dyer, E. S. & Bejerano, G. WhichTF is functionally important in your open chromatin data? PLoS Comput Biol 18, e1010378 (2022). https://doi.org/10.1371/journal.pcbi.1010378

- 24	Jolma, A. et al. DNA-binding specificities of human transcription factors. Cell 152, 327-339 (2013). https://doi.org/10.1016/j.cell.2012.12.009

- 25	Slattery, M. et al. Absence of a simple code: how transcription factors read the genome. Trends Biochem Sci 39, 381-399 (2014). https://doi.org/10.1016/j.tibs.2014.07.002

- 26	Aibar, S. et al. SCENIC: single-cell regulatory network inference and clustering. Nat Methods 14, 1083-1086 (2017). https://doi.org/10.1038/nmeth.4463

- 27	Oki, S. et al. ChIP-Atlas: a data-mining suite powered by full integration of public ChIP-seq data. EMBO Rep 19 (2018). https://doi.org/10.15252/embr.201846255

- 28	Lu, Z., Xiao, X., Zheng, Q., Wang, X. & Xu, L. Assessing NGS-based computational methods for predicting transcriptional regulators with query gene sets. bioRxiv, 2024.2002.2001.578316 (2024). https://doi.org/10.1101/2024.02.01.578316

- 29	Tang, F. et al. Chromatin profiles classify castration-resistant prostate cancers suggesting therapeutic targets. Science (New York, N.Y.) 376, eabe1505 (2022). https://doi.org/10.1126/science.abe1505

- 30	McLean, C. Y. et al. GREAT improves functional interpretation of cis-regulatory regions. Nat Biotechnol 28, 495-501 (2010). https://doi.org/10.1038/nbt.1630

- 31	Creyghton, M. P. et al. Histone H3K27ac separates active from poised enhancers and predicts developmental state. Proceedings of the National Academy of Sciences of the United States of America 107, 21931-21936 (2010). https://doi.org/10.1073/pnas.1016071107

- 32	Yan, F., Powell, D. R., Curtis, D. J. & Wong, N. C. From reads to insight: a hitchhiker's guide to ATAC-seq data analysis. Genome Biology 21, 22 (2020). https://doi.org/10.1186/s13059-020-1929-3

- 33	Wang, S. et al. Modeling cis-regulation with a compendium of genome-wide histone H3K27ac profiles. Genome Research 26, 1417-1429 (2016). https://doi.org/10.1101/gr.201574.115

- 34	Kharchenko, P. V., Tolstorukov, M. Y. & Park, P. J. Design and analysis of ChIP-seq experiments for DNA-binding proteins. Nature Biotechnology 26, 1351-1359 (2008). https://doi.org/10.1038/nbt.1508

- 35	Xu, B. et al. Acute depletion of CTCF rewires genome-wide chromatin accessibility. Genome Biol 22, 244 (2021). https://doi.org/10.1186/s13059-021-02466-0

- 36	Deng, W. et al. Reactivation of developmentally silenced globin genes by forced chromatin looping. Cell 158, 849-860 (2014). https://doi.org/10.1016/j.cell.2014.05.050

- 37	Lee, K. et al. FOXA2 Is Required for Enhancer Priming during Pancreatic Differentiation. Cell Rep 28, 382-393.e387 (2019). https://doi.org/10.1016/j.celrep.2019.06.034

- 38	Gruber, S., Haering, C. H. & Nasmyth, K. Chromosomal cohesin forms a ring. Cell 112, 765-777 (2003). https://doi.org/10.1016/s0092-8674(03)00162-4

- 39	Haering, C. H., Löwe, J., Hochwagen, A. & Nasmyth, K. Molecular architecture of SMC proteins and the yeast cohesin complex. Mol Cell 9, 773-788 (2002). https://doi.org/10.1016/s1097-2765(02)00515-4

- 40	Kadauke, S. & Blobel, G. A. Chromatin loops in gene regulation. Biochim Biophys Acta 1789, 17-25 (2009). https://doi.org/10.1016/j.bbagrm.2008.07.002

- 41	Hansen, A. S., Cattoglio, C., Darzacq, X. & Tjian, R. Recent evidence that TADs and chromatin loops are dynamic structures. Nucleus 9, 20-32 (2018). https://doi.org/10.1080/19491034.2017.1389365

- 42	Nora, E. P. et al. Targeted Degradation of CTCF Decouples Local Insulation of Chromosome Domains from Genomic Compartmentalization. Cell 169, 930-944.e922 (2017). https://doi.org/10.1016/j.cell.2017.05.004

- 43	Rao, S. S. P. et al. Cohesin Loss Eliminates All Loop Domains. Cell 171, 305-320.e324 (2017). https://doi.org/10.1016/j.cell.2017.09.026

- 44	Leonard, M., Brice, M., Engel, J. D. & Papayannopoulou, T. Dynamics of GATA transcription factor expression during erythroid differentiation. Blood 82, 1071-1079 (1993). 

- 45	Cheng, Y. et al. Erythroid GATA1 function revealed by genome-wide analysis of transcription factor occupancy, histone modifications, and mRNA expression. Genome Res 19, 2172-2184 (2009). https://doi.org/10.1101/gr.098921.109

- 46	Maeda, T. et al. LRF is an essential downstream target of GATA1 in erythroid development and regulates BIM-dependent apoptosis. Dev Cell 17, 527-540 (2009). https://doi.org/10.1016/j.devcel.2009.09.005

- 47	Tallack, M. R. et al. A global role for KLF1 in erythropoiesis revealed by ChIP-seq in primary erythroid cells. Genome Res 20, 1052-1063 (2010). https://doi.org/10.1101/gr.106575.110

- 48	Siatecka, M. & Bieker, J. J. The multifunctional role of EKLF/KLF1 during erythropoiesis. Blood 118, 2044-2054 (2011). https://doi.org/10.1182/blood-2011-03-331371

- 49	Wang, H. et al. Pdx1 level defines pancreatic gene expression pattern and cell lineage differentiation. J Biol Chem 276, 25279-25286 (2001). https://doi.org/10.1074/jbc.M101233200

- 50	Watt, A. J., Zhao, R., Li, J. & Duncan, S. A. Development of the mammalian liver and ventral pancreas is dependent on GATA4. BMC Dev Biol 7, 37 (2007). https://doi.org/10.1186/1471-213x-7-37

- 51	Fujimoto, K. & Polonsky, K. S. Pdx1 and other factors that regulate pancreatic beta-cell survival. Diabetes Obes Metab 11 Suppl 4, 30-37 (2009). https://doi.org/10.1111/j.1463-1326.2009.01121.x

- 52	Allen, H. L. et al. GATA6 haploinsufficiency causes pancreatic agenesis in humans. Nat Genet 44, 20-22 (2011). https://doi.org/10.1038/ng.1035

- 53	Iwafuchi-Doi, M. & Zaret, K. S. Pioneer transcription factors in cell reprogramming. Genes Dev 28, 2679-2692 (2014). https://doi.org/10.1101/gad.253443.114

- 54	Norton, L. J. et al. KLF1 directly activates expression of the novel fetal globin repressor ZBTB7A/LRF in erythroid cells. Blood Adv 1, 685-692 (2017). https://doi.org/10.1182/bloodadvances.2016002303

- 55	Meyers, R. M. et al. Computational correction of copy number effect improves specificity of CRISPR-Cas9 essentiality screens in cancer cells. Nat Genet 49, 1779-1784 (2017). https://doi.org/10.1038/ng.3984

- 56	Hurtado, A., Holmes, K. A., Ross-Innes, C. S., Schmidt, D. & Carroll, J. S. FOXA1 is a key determinant of estrogen receptor function and endocrine response. Nature Genetics 43, 27-33 (2011). https://doi.org/10.1038/ng.730

- 57	Holst, F. et al. Estrogen receptor alpha (ESR1) gene amplification is frequent in breast cancer. Nature Genetics 39, 655-660 (2007). https://doi.org/10.1038/ng2006

- 58	Mohammed, H. et al. Progesterone receptor modulates ERα action in breast cancer. Nature 523, 313-317 (2015). https://doi.org/10.1038/nature14583

- 59	Puhr, M. et al. PIAS1 is a determinant of poor survival and acts as a positive feedback regulator of AR signaling through enhanced AR stabilization in prostate cancer. Oncogene 35, 2322-2332 (2016). https://doi.org/10.1038/onc.2015.292

- 60	Heinlein, C. A. & Chang, C. Androgen receptor in prostate cancer. Endocr Rev 25, 276-308 (2004). https://doi.org/10.1210/er.2002-0032

- 61	Ewing, C. M. et al. Germline mutations in HOXB13 and prostate-cancer risk. N Engl J Med 366, 141-149 (2012). https://doi.org/10.1056/NEJMoa1110000

- 62	Hart, A. H. et al. The pluripotency homeobox gene NANOG is expressed in human germ cell tumors. Cancer 104, 2092-2098 (2005). https://doi.org/10.1002/cncr.21435

- 63	Irie, N. et al. SOX17 is a critical specifier of human primordial germ cell fate. Cell 160, 253-268 (2015). https://doi.org/10.1016/j.cell.2014.12.013

- 64	Looijenga, L. H. et al. POU5F1 (OCT3/4) identifies cells with pluripotent potential in human germ cell tumors. Cancer Res 63, 2244-2250 (2003). 

- 65	Watson, P. A., Arora, V. K. & Sawyers, C. L. Emerging mechanisms of resistance to androgen receptor inhibitors in prostate cancer. Nat Rev Cancer 15, 701-711 (2015). https://doi.org/10.1038/nrc4016

- 66	Patel, H. K. & Bihani, T. Selective estrogen receptor modulators (SERMs) and selective estrogen receptor degraders (SERDs) in cancer treatment. Pharmacol Ther 186, 1-24 (2018). https://doi.org/10.1016/j.pharmthera.2017.12.012

- 67	Anzick, S. L. et al. AIB1, a steroid receptor coactivator amplified in breast and ovarian cancer. Science 277, 965-968 (1997). https://doi.org/10.1126/science.277.5328.965

- 68	Kim, M. R., Wu, M. J., Zhang, Y., Yang, J. Y. & Chang, C. J. TET2 directs mammary luminal cell differentiation and endocrine response. Nat Commun 11, 4642 (2020). https://doi.org/10.1038/s41467-020-18129-w

- 69	Durbin, A. D. et al. Selective gene dependencies in MYCN-amplified neuroblastoma include the core transcriptional regulatory circuitry. Nat Genet 50, 1240-1246 (2018). https://doi.org/10.1038/s41588-018-0191-z

- 70	Gryder, B. E. et al. Histone hyperacetylation disrupts core gene regulatory architecture in rhabdomyosarcoma. Nat Genet 51, 1714-1722 (2019). https://doi.org/10.1038/s41588-019-0534-4

- 71	Polson, N. G., Scott, J. G. & Windle, J. Bayesian Inference for Logistic Models Using Pólya–Gamma Latent Variables. Journal of the American Statistical Association 108, 1339-1349 (2013). https://doi.org/10.1080/01621459.2013.829001

